### PR TITLE
DOP-3552: Fix unit tests since Gatsby v5 upgrade

### PR DESCRIPTION
**Pull request by rayangler**
### Stories/Links:

DOP-3552

### Current Behavior:

(Using branches for `gatsbyv4` base branch)

[Landing](https://docs-mongodbcom-integration.corp.mongodb.com/master/landing/raymundrodriguez/gatsbyv4/)
[Server Manual](https://docs-mongodbcom-integration.corp.mongodb.com/master/docs/raymundrodriguez/gatsbyv4/)
[Atlas](https://docs-mongodbcom-integration.corp.mongodb.com/master/cloud-docs/raymundrodriguez/gatsbyv4/)

### Staging Links:

(No changes expected)

[Landing](https://docs-mongodbcom-integration.corp.mongodb.com/master/landing/raymundrodriguez/gatsbyv4-tests-1/index.html)
[Server Manual](https://docs-mongodbcom-integration.corp.mongodb.com/master/docs/raymundrodriguez/gatsbyv4-tests-1/index.html)
[Atlas](https://docs-mongodbcom-integration.corp.mongodb.com/master/cloud-docs/raymundrodriguez/gatsbyv4-tests-1/index.html)

### Testing Locally

Pull this branch and run the following, assuming you have `nvm` installed:

```
nvm use && npm ci --legacy-peer-deps && npm run test
```

Feel free to compare with the base `gatsbyv4` branch to see the differences between test runs.

### Notes:
* Updates dev dependencies and test configurations to allow unit tests to run properly. Some of the major issues revolved around the `jsdom` environment, which is not included by default with newer versions of `jest`.
* It's unclear what the root cause of the tests not running is. Presumably, it's a weird Node 18 thing, but upgrading certain dev dependencies seems to resolve things okay.
* Tests for `Code` and `CodeIO` continue to fail due to differences in snapshots. These snapshot differences are a result of the syntax highlighting issue, which is being investigated separately.

### CODE REVIEW

File: tests/testSetup.js [hunk 1]
```diff
@@ -37,3 +37,4 @@ Object.defineProperty(global.self, 'crypto', {
 
 window.matchMedia = () => ({ addListener: () => {}, removeListener: () => {} });
 window.scrollTo = () => {};
+global.fetch = jest.fn();
```
Comment (seungpark) [inline tests/testSetup.js:40]: thinking out loud: wondering why this wasn't an issue before?

File: tests/unit/__snapshots__/SearchResults.test.js.snap [hunk 2]
```diff
@@ -4180,6 +4188,10 @@ exports[`Search Results Page renders results from a given search term query para
     margin-bottom: 8px;
     margin-right: 0;
   }
+
+  .emotion-34>div>div {
```
Comment (seungpark) [inline tests/unit/__snapshots__/SearchResults.test.js.snap:4192]: minor. where are these emotion style diffs coming from?

**Review by seungpark — APPROVED** [review]

File: tests/testSetup.js (hunk 1 above)
Comment (rayangler) [reply tests/testSetup.js:40]: I couldn't find a conclusive answer, but I'm assuming it has something to do with Node 18 doing some [experimental things ](https://nodejs.org/en/blog/announcements/v18-release-announce#fetch-experimental) with `fetch`. There's also this [SO post](https://stackoverflow.com/a/75399601).

File: tests/unit/__snapshots__/SearchResults.test.js.snap (hunk 2 above)
Comment (rayangler) [reply tests/unit/__snapshots__/SearchResults.test.js.snap:4192]: I believe they may be appearing now with proper usage of `await tick()` in the updated test file, leading to application of emotion css properly? This CSS property is technically [supposed to be there](https://github.com/mongodb/snooty/blob/945ffeb659ba18f1626ca60f50e90d38860ce428/src/components/SearchResults/SearchResults.js#L126) but wasn't included in the original snapshot

**Pull request merged**